bundles/org.openhab.binding.icloud/src/test/java/org/openhab/binding/icloud/TestICloud.java

```diff
@@ -74,7 +74,9 @@ public void testAuth() throws IOException, InterruptedException, ICloudApiRespon
         service.authenticate(false);
         if (service.requires2fa()) {
             System.out.print("Code: ");
-            String code = new Scanner(System.in).nextLine();
+            Scanner in = new Scanner(System.in);
+            String code = in.nextLine();
+            in.close();
```

Comment:
System.in should not be closed. It is managed by the JVM.

Reply:
compiler warned about it, did some reading and yeah compilers are sometimes silly. Fixed by surpressing the warning.